panic on bc info

```
{"level":"info","ts":1597948538.394371,"caller":"blocksync/buffer.go:89","msg":"Successfully committed block.","ioAddr":"io1xj0u5n20tsqwxh5a3xdtmzuz9wasft0pqjrq8t","networkAddr":"0.0.0.0:4689","recvHeight":6532711,"confirmedHeight":6532710,"source":"blockBuffer","syncedHeight":6532711}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0xf3c93e]

goroutine 12789 [running]:
github.com/iotexproject/iotex-core/api.(*Server).putBlockMetaUpgradeByHeader(0xc000115800, 0x9eb10, 0xc0004d3970, 0xc0004d3970, 0x0, 0x0)
	/go/apps/iotex-core/api/api.go:1216 +0x2e
github.com/iotexproject/iotex-core/api.(*Server).getBlockMetasByHeader(0xc000115800, 0x9eb10, 0x7f7dd9c51f40, 0xc000422a80, 0xe165ee2ae0478c86)
	/go/apps/iotex-core/api/api.go:1232 +0x141
github.com/iotexproject/iotex-core/api.(*Server).getBlockMetas(0xc000115800, 0x9eb10, 0x1, 0x13160a0, 0x0, 0x168fe20)
	/go/apps/iotex-core/api/api.go:1168 +0x181
github.com/iotexproject/iotex-core/api.(*Server).GetBlockMetas(0xc000115800, 0x1678d40, 0xc000e88690, 0xc000e886c0, 0xc000115800, 0x7f7dd9c51f40, 0xc00002d440)
	/go/apps/iotex-core/api/api.go:237 +0x89
github.com/iotexproject/iotex-proto/golang/iotexapi._APIService_GetBlockMetas_Handler.func1(0x1678d40, 0xc000e88690, 0x1291ea0, 0xc000e886c0, 0x22, 0xc000bc5270, 0xc0007a3600, 0xc00135f9b0)
	/go/pkg/mod/github.com/iotexproject/iotex-proto@v0.3.2-0.20200729044038-c22fbb206571/golang/iotexapi/api.pb.go:3718 +0x86
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1(0x1678d40, 0xc000e88690, 0x1291ea0, 0xc000e886c0, 0xc0014e5a20, 0xc0014e5a40, 0xc000649a20, 0x4a1ee8, 0x1252340, 0xc000e88690)
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107 +0xad
github.com/iotexproject/iotex-proto/golang/iotexapi._APIService_GetBlockMetas_Handler(0x13160a0, 0xc000115800, 0x1678d40, 0xc000e88690, 0xc000bc4a00, 0xc0004240d0, 0x1678d40, 0xc000e88690, 0xc000463090, 0x8)
	/go/pkg/mod/github.com/iotexproject/iotex-proto@v0.3.2-0.20200729044038-c22fbb206571/golang/iotexapi/api.pb.go:3720 +0x14b
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00051e300, 0x1687400, 0xc000bc8600, 0xc0007a3600, 0xc00003c6f0, 0x1f5f8b0, 0x0, 0x0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.21.0/server.go:998 +0x46a
google.golang.org/grpc.(*Server).handleStream(0xc00051e300, 0x1687400, 0xc000bc8600, 0xc0007a3600, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.21.0/server.go:1278 +0xd3d
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0004626b0, 0xc00051e300, 0x1687400, 0xc000bc8600, 0xc0007a3600)
	/go/pkg/mod/google.golang.org/grpc@v1.21.0/server.go:717 +0xa1
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.21.0/server.go:715 +0xa1
```

--- a/api/api.go
+++ b/api/api.go
@@ -1168,13 +1168,8 @@ func (api *Server) getBlockMetas(start uint64, count uint64) (*iotexapi.GetBlock
 	}
 	var res []*iotextypes.BlockMeta
 	for height := start; height <= tipHeight && count > 0; height++ {
-		blockMeta, err := api.getBlockMetasByHeader(height)
-		if errors.Cause(err) == db.ErrNotExist {
-			blockMeta, err = api.getBlockMetasByBlock(height)
-			if err != nil {
-				return nil, err
-			}
-		} else if err != nil {
+		blockMeta, err := api.getBlockMetaByHeight(height)
+		if err != nil {
 			return nil, err
 		}
 		res = append(res, blockMeta)
@@ -1192,19 +1187,36 @@ func (api *Server) getBlockMeta(blkHash string) (*iotexapi.GetBlockMetasResponse
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-	blockMeta, err := api.getBlockMetaByHeader(hash)
-	if errors.Cause(err) == db.ErrNotExist {
-		blockMeta, err = api.getBlockMetaByBlock(hash)
-		if err != nil {
-			return nil, err
-		}
-	} else if err != nil {
+	blockMeta, err := api.getBlockMetaByHash(hash)
+	if err != nil {
 		return nil, err
 	}
 	return &iotexapi.GetBlockMetasResponse{
 		Total:    1,
 		BlkMetas: []*iotextypes.BlockMeta{blockMeta},
 	}, nil
+}
+
+// getBlockMetaByHeight gets block meta by height
+func (api *Server) getBlockMetaByHeight(height uint64) (*iotextypes.BlockMeta, error) {
+	if api.indexer != nil {
+		blockMeta, err := api.getBlockMetasByHeader(height)
+		if errors.Cause(err) != db.ErrNotExist {
+			return blockMeta, err
+		}
+	}
+	return api.getBlockMetasByBlock(height)
+}
+
+// getBlockMetaByHash gets block meta by hash
+func (api *Server) getBlockMetaByHash(h hash.Hash256) (*iotextypes.BlockMeta, error) {
+	if api.indexer != nil {
+		blockMeta, err := api.getBlockMetaByHeader(h)
+		if errors.Cause(err) != db.ErrNotExist {
+			return blockMeta, err
+		}
+	}
+	return api.getBlockMetaByBlock(h)
 }
 
 // putBlockMetaUpgradeByBlock puts numActions and transferAmount for blockmeta by block

